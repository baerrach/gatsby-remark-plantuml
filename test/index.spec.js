const path = require(`path`)

const testRemarkPlugin = require(`./test-remark-plugin`)

describe(`remark plantuml plugin`, () => {
  beforeEach(() => {
    jest.resetModules()
  })

  describe(`ignores codeblocks that are not plantuml`, () => {
    it(`with lang=''`, async () => {
      const code = `\`\`\`\n// Fake\n\`\`\``
      await testRemarkPlugin.testPlugin({
        code,
      })
    })

    it(`with lang='js'`, async () => {
      const code = `\`\`\`js\n// Fake\n\`\`\``

      await testRemarkPlugin.testPlugin({
        code,
      })
    })
  })

  describe(`configuration`, () => {
    it(`plantumljar as absolute path`, async () => {
      const code = `
\`\`\`plantuml
@startuml
Alice -> Bob: Authentication Request
Bob --> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <-- Bob: Another authentication Response
@enduml
\`\`\`
    `
      await testRemarkPlugin.testPlugin({
        code,
        undefined,
        options: {
          plantumljar: path.resolve(
            __dirname,
            `../lib/plantuml-jar-mit-1.2020.15/plantuml.jar`
          ),
        },
      })
    })

    it(`plantumljar as relative path`, async () => {
      const code = `
\`\`\`plantuml
@startuml
Alice -> Bob: Authentication Request
Bob --> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <-- Bob: Another authentication Response
@enduml
\`\`\`
    `
      await testRemarkPlugin.testPlugin({
        code,
        undefined,
        options: {
          plantumljar: `./lib/plantuml-jar-mit-1.2020.15/plantuml.jar`,
        },
      })
    })
  })
})
