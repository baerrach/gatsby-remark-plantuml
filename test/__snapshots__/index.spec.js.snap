// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`remark plantuml plugin configuration plantumljar as absolute path 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 10,
          "offset": 203,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" xlink=\\"http://www.w3.org/1999/xlink\\" contentscripttype=\\"application/ecmascript\\" contentstyletype=\\"text/css\\" height=\\"216px\\" preserveAspectRatio=\\"none\\" style=\\"max-width: 100%; height: auto;\\" version=\\"1.1\\" viewBox=\\"0 0 295 216\\" width=\\"295px\\" zoomAndPan=\\"magnify\\"><defs><filter height=\\"300%\\" id=\\"fn3yvorrca0j1\\" width=\\"300%\\" x=\\"-1\\" y=\\"-1\\"><feGaussianBlur result=\\"blurOut\\" stdDeviation=\\"2.0\\"/><feColorMatrix in=\\"blurOut\\" result=\\"blurOut2\\" type=\\"matrix\\" values=\\"0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0\\"/><feOffset dx=\\"4.0\\" dy=\\"4.0\\" in=\\"blurOut2\\" result=\\"blurOut3\\"/><feBlend in=\\"SourceGraphic\\" in2=\\"blurOut3\\" mode=\\"normal\\"/></filter></defs><g><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 5.0,5.0;\\" x1=\\"33\\" x2=\\"33\\" y1=\\"38.4883\\" y2=\\"175.7305\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 5.0,5.0;\\" x1=\\"266.5\\" x2=\\"266.5\\" y1=\\"38.4883\\" y2=\\"175.7305\\"/><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"47\\" x=\\"8\\" y=\\"3\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"33\\" x=\\"15\\" y=\\"23.5352\\">Alice</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"47\\" x=\\"8\\" y=\\"174.7305\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"33\\" x=\\"15\\" y=\\"195.2656\\">Alice</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"40\\" x=\\"244.5\\" y=\\"3\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"26\\" x=\\"251.5\\" y=\\"23.5352\\">Bob</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"40\\" x=\\"244.5\\" y=\\"174.7305\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"26\\" x=\\"251.5\\" y=\\"195.2656\\">Bob</text><polygon fill=\\"#A80036\\" points=\\"254.5,65.7988,264.5,69.7988,254.5,73.7988,258.5,69.7988\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0;\\" x1=\\"33.5\\" x2=\\"260.5\\" y1=\\"69.7988\\" y2=\\"69.7988\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"147\\" x=\\"40.5\\" y=\\"65.0566\\">Authentication Request</text><polygon fill=\\"#A80036\\" points=\\"44.5,95.1094,34.5,99.1094,44.5,103.1094,40.5,99.1094\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 2.0,2.0;\\" x1=\\"38.5\\" x2=\\"265.5\\" y1=\\"99.1094\\" y2=\\"99.1094\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"157\\" x=\\"50.5\\" y=\\"94.3672\\">Authentication Response</text><polygon fill=\\"#A80036\\" points=\\"254.5,124.4199,264.5,128.4199,254.5,132.4199,258.5,128.4199\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0;\\" x1=\\"33.5\\" x2=\\"260.5\\" y1=\\"128.4199\\" y2=\\"128.4199\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"199\\" x=\\"40.5\\" y=\\"123.6777\\">Another authentication Request</text><polygon fill=\\"#A80036\\" points=\\"44.5,153.7305,34.5,157.7305,44.5,161.7305,40.5,157.7305\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 2.0,2.0;\\" x1=\\"38.5\\" x2=\\"265.5\\" y1=\\"157.7305\\" y2=\\"157.7305\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"209\\" x=\\"50.5\\" y=\\"152.9883\\">Another authentication Response</text><!--
@startuml
Alice -> Bob: Authentication Request
Bob - -> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <- - Bob: Another authentication Response
@enduml


--></g></svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 5,
      "line": 11,
      "offset": 208,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark plantuml plugin configuration plantumljar as relative path 1`] = `
Object {
  "children": Array [
    Object {
      "children": undefined,
      "lang": undefined,
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 10,
          "offset": 203,
        },
        "indent": Array [
          1,
          1,
          1,
          1,
          1,
          1,
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 2,
          "offset": 1,
        },
      },
      "type": "html",
      "value": "<svg xmlns=\\"http://www.w3.org/2000/svg\\" xlink=\\"http://www.w3.org/1999/xlink\\" contentscripttype=\\"application/ecmascript\\" contentstyletype=\\"text/css\\" height=\\"216px\\" preserveAspectRatio=\\"none\\" style=\\"max-width: 100%; height: auto;\\" version=\\"1.1\\" viewBox=\\"0 0 295 216\\" width=\\"295px\\" zoomAndPan=\\"magnify\\"><defs><filter height=\\"300%\\" id=\\"fn3yvorrca0j1\\" width=\\"300%\\" x=\\"-1\\" y=\\"-1\\"><feGaussianBlur result=\\"blurOut\\" stdDeviation=\\"2.0\\"/><feColorMatrix in=\\"blurOut\\" result=\\"blurOut2\\" type=\\"matrix\\" values=\\"0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0 .4 0\\"/><feOffset dx=\\"4.0\\" dy=\\"4.0\\" in=\\"blurOut2\\" result=\\"blurOut3\\"/><feBlend in=\\"SourceGraphic\\" in2=\\"blurOut3\\" mode=\\"normal\\"/></filter></defs><g><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 5.0,5.0;\\" x1=\\"33\\" x2=\\"33\\" y1=\\"38.4883\\" y2=\\"175.7305\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 5.0,5.0;\\" x1=\\"266.5\\" x2=\\"266.5\\" y1=\\"38.4883\\" y2=\\"175.7305\\"/><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"47\\" x=\\"8\\" y=\\"3\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"33\\" x=\\"15\\" y=\\"23.5352\\">Alice</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"47\\" x=\\"8\\" y=\\"174.7305\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"33\\" x=\\"15\\" y=\\"195.2656\\">Alice</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"40\\" x=\\"244.5\\" y=\\"3\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"26\\" x=\\"251.5\\" y=\\"23.5352\\">Bob</text><rect fill=\\"#FEFECE\\" filter=\\"url(#fn3yvorrca0j1)\\" height=\\"30.4883\\" style=\\"stroke: #A80036; stroke-width: 1.5;\\" width=\\"40\\" x=\\"244.5\\" y=\\"174.7305\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"14\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"26\\" x=\\"251.5\\" y=\\"195.2656\\">Bob</text><polygon fill=\\"#A80036\\" points=\\"254.5,65.7988,264.5,69.7988,254.5,73.7988,258.5,69.7988\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0;\\" x1=\\"33.5\\" x2=\\"260.5\\" y1=\\"69.7988\\" y2=\\"69.7988\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"147\\" x=\\"40.5\\" y=\\"65.0566\\">Authentication Request</text><polygon fill=\\"#A80036\\" points=\\"44.5,95.1094,34.5,99.1094,44.5,103.1094,40.5,99.1094\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 2.0,2.0;\\" x1=\\"38.5\\" x2=\\"265.5\\" y1=\\"99.1094\\" y2=\\"99.1094\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"157\\" x=\\"50.5\\" y=\\"94.3672\\">Authentication Response</text><polygon fill=\\"#A80036\\" points=\\"254.5,124.4199,264.5,128.4199,254.5,132.4199,258.5,128.4199\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0;\\" x1=\\"33.5\\" x2=\\"260.5\\" y1=\\"128.4199\\" y2=\\"128.4199\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"199\\" x=\\"40.5\\" y=\\"123.6777\\">Another authentication Request</text><polygon fill=\\"#A80036\\" points=\\"44.5,153.7305,34.5,157.7305,44.5,161.7305,40.5,157.7305\\" style=\\"stroke: #A80036; stroke-width: 1.0;\\"/><line style=\\"stroke: #A80036; stroke-width: 1.0; stroke-dasharray: 2.0,2.0;\\" x1=\\"38.5\\" x2=\\"265.5\\" y1=\\"157.7305\\" y2=\\"157.7305\\"/><text fill=\\"#000000\\" font-family=\\"sans-serif\\" font-size=\\"13\\" lengthAdjust=\\"spacingAndGlyphs\\" textLength=\\"209\\" x=\\"50.5\\" y=\\"152.9883\\">Another authentication Response</text><!--
@startuml
Alice -> Bob: Authentication Request
Bob - -> Alice: Authentication Response

Alice -> Bob: Another authentication Request
Alice <- - Bob: Another authentication Response
@enduml


--></g></svg>",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 5,
      "line": 11,
      "offset": 208,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark plantuml plugin ignores codeblocks that are not plantuml with lang='' 1`] = `
Object {
  "children": Array [
    Object {
      "lang": null,
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 15,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "code",
      "value": "// Fake",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 15,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;

exports[`remark plantuml plugin ignores codeblocks that are not plantuml with lang='js' 1`] = `
Object {
  "children": Array [
    Object {
      "lang": "js",
      "meta": null,
      "position": Position {
        "end": Object {
          "column": 4,
          "line": 3,
          "offset": 17,
        },
        "indent": Array [
          1,
          1,
        ],
        "start": Object {
          "column": 1,
          "line": 1,
          "offset": 0,
        },
      },
      "type": "code",
      "value": "// Fake",
    },
  ],
  "position": Object {
    "end": Object {
      "column": 4,
      "line": 3,
      "offset": 17,
    },
    "start": Object {
      "column": 1,
      "line": 1,
      "offset": 0,
    },
  },
  "type": "root",
}
`;
